backend/infrahub/core/diff/query_parser.py

```diff
@@ -583,14 +604,25 @@ def _update_attribute_property(
         )
 
     def _get_diff_relationship(
-        self, diff_node: DiffNodeIntermediate, relationship_schema: RelationshipSchema
+        self,
+        diff_node: DiffNodeIntermediate,
+        relationship_schema: RelationshipSchema,
+        database_path: DatabasePath,
     ) -> DiffRelationshipIntermediate:
         diff_relationship = diff_node.relationships_by_name.get(relationship_schema.name)
         if not diff_relationship:
+            node_field_specifier = NodeFieldSpecifier(
+                node_uuid=diff_node.uuid, field_name=relationship_schema.get_identifier()
+            )
+            branch_name = database_path.deepest_branch
+            from_time = self.from_time
+            if branch_name == self.base_branch_name and node_field_specifier in self.get_new_node_field_specifiers():
+                from_time = self.diff_branched_from_time
             diff_relationship = DiffRelationshipIntermediate(
                 name=relationship_schema.name,
                 cardinality=relationship_schema.cardinality,
                 identifier=relationship_schema.get_identifier(),
+                from_time=from_time,
```

Comment:
we use the `from_time` here and above in `_get_diff_attribute` when determining the action (ADDED/UPDATED/REMOVED) for a given database path. when we are getting nodes on the base branch (always `main` for now), we retrieve both completely new nodes that are not yet part of this diff and updated nodes that are already part of this diff. we need to use different `from_time`s when determining their actions and this is how we accomplish that